File: string-paradigm-expression_2/src/test/java/org/jesperancinha/string/paradigm/original/expression/SpellingCheckImplTest.java
package org.jesperancinha.string.paradigm.original.expression;

import static com.google.common.truth.Truth.assertThat;

import java.util.List;

import org.junit.Test;

public class SpellingCheckImplTest {

	private static final String _1 = "1'";
	private static final String _2 = "2'";
	private static final String _3 = ",3";
	private static final String ALGARVE = "Algarve";
	private static final String EXTREMADURA = "Tras Os Montes E Alto Douro, , extremadura";
	private static final String RIBATEJO = "  Ribatejo";
	private static final String MINHO = "minho";

	@Test
	public void testCheckSpelling() {
		String[] regions = { ALGARVE, MINHO, RIBATEJO, EXTREMADURA };
		String[] assignedNumbers = { _1, _2, _3 };
		SpellingCheckImpl check = new SpellingCheckImpl();
		for (String region : regions) {
			check.checkSpelling(region);
		}
		for (String number : assignedNumbers) {
			check.checkSpelling(number);
		}

		final List<String> resultSpellingBeginCapitalRegions = check.getSpellingBeginCapitalRegions();
		assertThat(resultSpellingBeginCapitalRegions.get(0)).startsWith(ALGARVE);
		assertThat(resultSpellingBeginCapitalRegions.get(1)).startsWith(RIBATEJO);

		final List<String> resultSpellingDoubleCommaRegions = check.getSpellingDoubleCommaRegions();
		assertThat(resultSpellingDoubleCommaRegions).hasSize(1);
		for (String stringResult : resultSpellingDoubleCommaRegions) {
			assertThat(stringResult).startsWith(EXTREMADURA);
		}

		final List<String> resultSpellingWhiteSpacesRegions = check.getSpellingWhiteSpacesRegions();
		assertThat(resultSpellingWhiteSpacesRegions).hasSize(1);
		for (String stringResult : resultSpellingWhiteSpacesRegions) {
			assertThat(stringResult).startsWith(RIBATEJO);
		}
	}
}

File: string-paradigm-expression_2/src/test/kotlin/org/jesperancinha/string/paradigm/original/expression/SpellingCheckImplTest.kt
package org.jesperancinha.string.paradigm.original.expression

import io.kotest.matchers.collections.shouldHaveSize
import io.kotest.matchers.string.shouldStartWith
import org.junit.jupiter.api.Test

class SpellingCheckImplTest {
    @Test
    fun testCheckSpelling() {
        val regions = arrayOf(ALGARVE, MINHO, RIBATEJO, EXTREMADURA)
        val assignedNumbers = arrayOf(_1, _2, _3)
        val check = SpellingCheckImpl()
        for (region in regions) {
            check.checkSpelling(region)
        }
        for (number in assignedNumbers) {
            check.checkSpelling(number)
        }
        val resultSpellingBeginCapitalRegions = check.spellingBeginCapitalRegions
        resultSpellingBeginCapitalRegions[0].shouldStartWith(ALGARVE)
        resultSpellingBeginCapitalRegions[1].shouldStartWith(RIBATEJO)
        val resultSpellingDoubleCommaRegions = check.spellingDoubleCommaRegions
        resultSpellingDoubleCommaRegions.shouldHaveSize(1)
        for (stringResult in resultSpellingDoubleCommaRegions) {
            stringResult.shouldStartWith(EXTREMADURA)
        }
        val resultSpellingWhiteSpacesRegions = check.spellingWhiteSpacesRegions
        resultSpellingWhiteSpacesRegions.shouldHaveSize(1)
        for (stringResult in resultSpellingWhiteSpacesRegions) {
            stringResult.shouldStartWith(RIBATEJO)
        }
    }

    companion object {
        private const val _1 = "1'"
        private const val _2 = "2'"
        private const val _3 = ",3"
        private const val ALGARVE = "Algarve"
        private const val EXTREMADURA = "Tras Os Montes E Alto Douro, , extremadura"
        private const val RIBATEJO = "  Ribatejo"
        private const val MINHO = "minho"
    }
}